### PR TITLE
Suggestion review form

### PR DESCRIPTION
**Pull request by phlax**
adds a suggestion review form and corresponding jsonview

will also remove the commentform from SuggestionReview tool

### CODE REVIEW

File: pootle/apps/pootle_store/views.py [hunk 1]
```diff
@@ -16,32 +16,37 @@
...
 from django.utils import timezone
 from django.utils.functional import cached_property
 from django.utils.lru_cache import lru_cache
 from django.utils.translation import to_locale
 from django.utils.translation.trans_real import parse_accept_lang_header
 from django.views.decorators.http import require_http_methods
+from django.views.generic import FormView
 
+from pootle.core.views.decorators import (
+    requires_permission, set_permissions)
```
Comment (unho) [inline pootle/apps/pootle_store/views.py]: Nit: single line import.

File: pootle/apps/pootle_store/forms.py [hunk 2]
```diff
@@ -502,3 +504,59 @@ def clean_path(self):
...
+    def review(self):
+        return review.get(self.target_object.__class__)(
+            [self.target_object],
+            self.cleaned_data["user"],
+            review_type=self.review_type)
+
+    def clean_action(self):
+        if self.target_object.state.name != "pending":
+            raise forms.ValidationError(
+                "Suggestion '%s' cannot be accepted/rejected twice!"
```
Comment (unho) [inline pootle/apps/pootle_store/forms.py]: I think this error gets exposed in UI, so needs to be localizable.

File: pootle/apps/pootle_store/forms.py [hunk 3]
```diff
@@ -502,3 +504,59 @@ def clean_path(self):
...
+            "view"
+            if self_review
+            else "review")
+        has_permission = check_user_permission(
+            self.cleaned_data["user"],
+            permission,
+            self.target_object.unit.store.parent)
+        if not has_permission:
+            raise forms.ValidationError(
+                _("Insufficient rights to access this page."))
```
Comment (unho) [inline pootle/apps/pootle_store/forms.py]: Shouldn't this be checked at view level?

Comment (phlax) [inline pootle/apps/pootle_store/forms.py]: no - because it uses the cleaned_data

File: pootle/apps/pootle_store/views.py [hunk 4]
```diff
@@ -618,62 +623,72 @@ def suggest(request, unit, **kwargs_):
...
+        ctx = super(UnitSuggestionJSON, self).get_context_data(*args, **kwargs)
+        form = ctx["form"]
+        if form.is_valid():
+            result = dict(
+                udbid=form.target_object.unit.id,
+                sugid=form.target_object.id,
+                user_score=self.request.user.public_score)
+            if form.cleaned_data["action"] == "accept":
+                result.update(
+                    dict(user_score=self.request.user.public_score,
```
Comment (unho) [inline pootle/apps/pootle_store/views.py]: `user_score` is repeated.

**Comment by unho** [comment]
lgtm, after nits.

**Pull request merged**